pandas/core/reshape/merge.py

```diff
@@ -1607,6 +1608,18 @@ def _factorize_keys(lk, rk, sort=True):
 
         lk = ensure_int64(lk.codes)
         rk = ensure_int64(rk)
+    elif (is_period_dtype(lk) and
```

Comment:
i think you can remove the period_dtype checks as these are now extension arrays

Reply:
Removing period_dtype checks would fails test_merge_on_period in test_merge.py.